File: Frontend/src/types/Mensaje.ts
export type CanalComunicacion = 'general' | 'arquitectura' | 'backend' | 'frontend' | 'qa' | 'directo'
export type EtiquetaMensaje = 'PREGUNTA' | 'APROBACION' | 'SEGURIDAD' | 'ACTUALIZACION' | 'BUG' | 'OK' | 'TAREA' | 'SISTEMA'

export interface Mensaje {
  id: string
  proyectoId: string
  canal: CanalComunicacion
  agenteOrigen: string
  agenteDestino: string
  etiqueta: EtiquetaMensaje
  contenido: string
  marcaTiempo: string
  esTyping: boolean
}

export interface EventoTyping {
  agente: string
  canal: CanalComunicacion
  escribiendo: boolean
}

export const COLOR_ETIQUETA: Record<EtiquetaMensaje, string> = {
  PREGUNTA: '#00d4ff',
  APROBACION: '#00ff88',
  SEGURIDAD: '#ff4466',
  ACTUALIZACION: '#ffd700',
  BUG: '#ff4466',
  OK: '#00ff88',
  TAREA: '#7c3aed',
  SISTEMA: '#6b7280',
}

File: Backend/models/mensaje.py
from enum import Enum
from pydantic import BaseModel
from datetime import datetime
from .agente import RolAgente


class CanalComunicacion(str, Enum):
    GENERAL = "general"
    ARQUITECTURA = "arquitectura"
    BACKEND = "backend"
    FRONTEND = "frontend"
    QA = "qa"
    DIRECTO = "directo"


class EtiquetaMensaje(str, Enum):
    PREGUNTA = "PREGUNTA"
    APROBACION = "APROBACION"
    SEGURIDAD = "SEGURIDAD"
    ACTUALIZACION = "ACTUALIZACION"
    BUG = "BUG"
    OK = "OK"
    TAREA = "TAREA"
    SISTEMA = "SISTEMA"


class Mensaje(BaseModel):
    id: str
    proyecto_id: str
    canal: CanalComunicacion
    agente_origen: str  # RolAgente o "sistema"
    agente_destino: str  # RolAgente o "todos"
    etiqueta: EtiquetaMensaje
    contenido: str
    marca_tiempo: datetime
    es_typing: bool = False

    class Config:
        from_attributes = True


class EventoWebSocket(BaseModel):
    tipo: str  # "mensaje" | "typing" | "estado_agente" | "proyecto_listo"
    datos: dict
